src/cpp/desktop/DesktopMain.cpp

```diff
@@ -298,7 +298,7 @@ int main(int argc, char* argv[])
          // get filename from command line arguments
          if (pApp->arguments().size() > 1)
          {
-            QString arg = pApp->arguments().last();
+            QString arg = pApp->arguments().at(1);
```

Comment:
This still relies on the file argument being in a particular position in order to be respected. I think this might be difficult to remember when invoking from elsewhere (we already broke it once!). Do you think it'd be worth removing the positional dependence by looking at the whole argument list?

Reply:
I think is fine for some parameters to be positional, positional at the end is hard, but we can assume that `rstudio <file-to-open> [--run-diagnostics] [--disable-renderer-accessibility] [--disable-web-security]` is the expected launching call. We could add `--help` message as some point.

Reply:
FYI, Pro desktop does have --help (see DesktopOptionsPro.cpp).